### ISSUE TITLE
how to analyze soft file on the local disk

### ISSUE THREAD

**Issue by wymin576** [seandavi/GEOquery#155]
I found that most help files for the package focus on automatically downloading the soft file from GEO. However, the internet in China is unstable.  As a result, I always download the soft file manually and then read it using the function 'getGEO'. My code is as follow,

```
gset <- getGEO(filename = 'D:/Bioinformatics/GSE74089/GSE74089_family.soft.gz',destdir = tempdir(),

               GSEMatrix = TRUE,AnnotGPL = TRUE,getGPL = TRUE)
```
I found the gene expression data is distributed in multiple samples, unlike the entire object from automatically downloaded. Here there are eight gene expression data frames. Shall I merge the eight data frames? I believe that there is an easy method to solve the issue. Can you recommend an introduction to analyze the GEO soft file on the local disk ?

Thank you in advance!

**Comment by DarioS** [comment]
I also desire this. It is clunky in the current implementation. Basically, afterwards, you need to do:

```r
cohortMatrix <- sapply(dataset$data@gsms, function(patient) patient@dataTable@table[, 2])
rownames(cohortMatrix) <- dataset$data@gsms[[1]]@dataTable@table[["ID_REF"]]
```

which should be standard package functionality. Also `?GSE-class` has woeful documentation. No slots, no accessors.

![Image](https://github.com/user-attachments/assets/214823af-c8b3-4914-b3d0-b23d99b32f2c)

**Comment by seandavi** [comment]
Sorry for the ancient documentation. Those classes and the workflow associated with soft files have largely been superseded by the GSEMatrix format files that GEO now produces. Here is the preferred workflow with equivalent information:

```
> download.file('https://ftp.ncbi.nlm.nih.gov/geo/series/GSE74nnn/GSE74089/matrix/GSE74089_series_matrix.txt.gz','GSE74089_series_matrix.txt.gz')
trying URL 'https://ftp.ncbi.nlm.nih.gov/geo/series/GSE74nnn/GSE74089/matrix/GSE74089_series_matrix.txt.gz'
Content type 'application/x-gzip' length 1203606 bytes (1.1 MB)
==================================================
downloaded 1.1 MB

> a = getGEO(filename='GSE74089_series_matrix.txt.gz')
> a
ExpressionSet (storageMode: lockedEnvironment)
assayData: 34183 features, 8 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: GSM1909502 GSM1909503 ... GSM1909509 (8 total)
  varLabels: title geo_accession ... tissue:ch1 (33 total)
  varMetadata: labelDescription
featureData
  featureNames: (+)E1A_r60_1 (+)E1A_r60_3 ... GE_BrightCorner (34183 total)
  fvarLabels: ID SPOT_ID ... SEQUENCE (17 total)
  fvarMetadata: Column Description labelDescription
experimentData: use 'experimentData(object)'
  pubMedIds: 39253583 
Annotation: GPL13497
```

I should definitely clean up and modernize the vignette. Note that the ExpressionSet can be converted to a SummarizedExperiment for a more modern experience.

**Comment by DarioS** [comment]
Ah, good to know about. How about RNA-seq abundances though, e.g. [GSE142667](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE142667)? Not in the SOFT nor Series Matrix.

**Comment by seandavi** [comment]
Yeah, that is covered by some new work in collaboration with the GEO folks. I have yet to find time to add it to the vignette. 

```r
library(GEOquery)
gse = getRNASeqData('GSE142667')
gse
```

```r
class: SummarizedExperiment 
dim: 39376 56 
metadata(5): experimentData annotation protocolData genomeInfo created_at
assays(1): counts
rownames(39376): 100287102 653635 ... 4576 4571
rowData names(18): GeneID Symbol ... GOProcess GOComponent
colnames(56): GSM4237439 GSM4237440 ... GSM4237494 GSM4237495
colData names(41): title geo_accession ... cell.type.ch1 diagnosis.ch1
```

Small note that if you want to try this immediately, you'll want to install from github or wait until the devel builds complete in about 48 hours. The functionality is in RELEASE as well, but there is a bug in sample id matching when GEO drops samples for QC reasons.

**Comment by DarioS** [comment]
I am creating a Shiny app. Will there be a wrapper function that auto-detects microarray or RNA-seq GSE?

**Comment by seandavi** [comment]
> I am creating a Shiny app. Will there be a wrapper function that auto-detects microarray or RNA-seq GSE?

The edge case is when there is both submitter-supplied data and GEO-calculated data. Would a parameter in getGEO be enough to specify the behavior when both are available?

**Comment by DarioS** [comment]
Ah, I did not consider that scenario.

**Comment by seandavi** [comment]
Just FYI, the RNA-seq quantification from NCBI is now documented: https://seandavi.github.io/GEOquery/articles/GEOquery.html#rna-seq-quantifications-ncbis-solution-to-reanalysis-challenges
